### DISCUSSION TITLE
Referencing other parameters results in out of order rendering

### DISCUSSION THREAD

**Discussion by jmthomas** [lsegal/yard#1559] [Ideas]
I use the ability to reference other parameters in conjunction with defining unique parameters. This works great for documenting inheritance chains. However, I've noticed that when doing this, the unique parameters are always rendered first in the documentation. I assume this is probably because of how you lookup the #see references and fill those in during another pass. Nothing serious but a little confusing because the parameter list doesn't match the method definition.

## Steps to reproduce
```
class MyYard
  # @param value [String] The string
  def base(value); end
  # @param value (see #base)
  # @param value2 [String] The other string
  def base2(value, value2); end
end
```
## Actual Output
![image](https://cloud.githubusercontent.com/assets/176403/21783228/a6c6ef02-d673-11e6-9ad0-af1ee5d71e87.png)

## Expected Output
Expected value to come before value2 in the parameter list.

## Environment details:
* OS: [Windows 7]
* Ruby version 2.2.3
* YARD version 0.9.5

I have read the [Contributing Guide][contrib].
[contrib]: https://github.com/lsegal/yard/blob/master/CONTRIBUTING.md

**Comment by lsegal** [comment]
We could probably sort the params by their placement in the method signature, if provided. Patches accepted for this.

**Comment by jmthomas** [comment]
Opened PR #1065
